### PR TITLE
Make a change to sc add gcp broker such that we return error when existing broker is EAP

### PR DESCRIPTION
**Pull request by deepika-chander**

### CODE REVIEW

File: installer/pkg/broker-cli/client/adapter/httpadapter.go [hunk 1]
```diff
@@ -39,6 +40,32 @@ type CreateBrokerParams struct {
 	Title   string
 }
 
+// GetCatalogParams is used as input to GetCatalog.
+type GetCatalogParams struct {
+	// Server is URL of the broker.
+	Server string
```
Comment (martinmaly) [inline installer/pkg/broker-cli/client/adapter/httpadapter.go]: consider URL to match CreateBrokerParams

File: installer/pkg/broker-cli/client/adapter/httpadapter.go [hunk 2]
```diff
@@ -39,6 +40,32 @@ type CreateBrokerParams struct {
...
+}
+
+// GetCatalogResult is output of successful GetCatalog request.
+type GetCatalogResult struct {
+	Services []Service
```
Comment (martinmaly) [inline installer/pkg/broker-cli/client/adapter/httpadapter.go]: `json:"...."` ?

File: installer/pkg/cmd/gcp_broker.go [hunk 3]
```diff
@@ -236,17 +268,38 @@ func getOrCreateVirtualBroker(projectID, brokerName, brokerTitle string) (*virtu
...
+	if errCode == http.StatusConflict {
+		fmt.Printf("Broker %q, already exists\n", brokerName)
+		res, err := brokerClient.GetCatalog(&adapter.GetCatalogParams{
+			Server: brokerURL,
+		})
+		if err != nil {
+			return nil, fmt.Errorf("Invalid broker %q, error getting catalog for broker: %v\n", brokerURL, err)
+		}
+
+		isEAPBroker := false
```
Comment (martinmaly) [inline installer/pkg/cmd/gcp_broker.go]: consider hoisting some part of the logic between 279-299 to a separate function.
Good candidate may be lines 288-299 or thereabouts.

File: installer/pkg/cmd/gcp_broker.go [hunk 4]
```diff
@@ -236,17 +268,38 @@ func getOrCreateVirtualBroker(projectID, brokerName, brokerTitle string) (*virtu
...
+		for _, svc := range res.Services {
+			if _, ok := eapServiceGuids[svc.ID]; ok {
+				isEAPBroker = true
+				break
+			}
+		}
+
+		if isEAPBroker {
+			return nil, fmt.Errorf("Existing broker is an EAP broker, please delete broker using broker-cli, and rerun \"sc add-gcp-broker\" to get Beta broker\n")
```
Comment (martinmaly) [inline installer/pkg/cmd/gcp_broker.go]: Your existing broker is deprecated. Please delete your broker using broker-cli and re-run sc add-gcp-broker.

File: installer/pkg/broker-cli/client/adapter/httpadapter.go (hunk 1 above)
Comment (deepika-chander) [reply installer/pkg/broker-cli/client/adapter/httpadapter.go]: done

File: installer/pkg/cmd/gcp_broker.go (hunk 3 above)
Comment (deepika-chander) [reply installer/pkg/cmd/gcp_broker.go]: done.

File: installer/pkg/cmd/gcp_broker.go (hunk 4 above)
Comment (deepika-chander) [reply installer/pkg/cmd/gcp_broker.go]: done

File: installer/pkg/cmd/gcp_broker.go [hunk 5]
```diff
@@ -61,6 +61,38 @@ var (
 		"sqladmin.googleapis.com",
 		"storage-api.googleapis.com",
 	}
+
+	eapServiceGuids = map[string]bool{
```
[the remark this answers is not in the document]
Comment (deepika-chander) [reply installer/pkg/cmd/gcp_broker.go]: done.

File: installer/pkg/broker-cli/client/adapter/httpadapter.go (hunk 2 above)
Comment (deepika-chander) [reply installer/pkg/broker-cli/client/adapter/httpadapter.go]: done

**Pull request merged**